lib/puppet/node.rb

```diff
@@ -27,10 +27,10 @@ def initialize_from_hash(data)
     @name       = data['name']       || (raise ArgumentError, _("No name provided in serialized data"))
     @classes    = data['classes']    || []
     @parameters = data['parameters'] || {}
-    env_name = data['environment']
+    env_name = data['environment'] || @parameters[ENVIRONMENT]
```

Comment:
Note for @jhelwig and myself: I can't figure out how to get a failing spec test after reversing this line's change, so I am not sure it is necessary/impactful.  Seems like the node_spec.rb test "uses the environment parameter when provided" or a nearby test should be sensitive.

Reply:
Talked with @kris-bosland and remembered why this line is here. It's mostly as a placeholder for when the `environment` method gets cleaned up, but is also to help figure out the intended precedence of the various ways to set what environment a `Node` has by being explicit here.